pkg/inputs/mongooplog/position_value.go

```
@@ -69,29 +65,25 @@ func SetupInitialPosition(cache position_store.PositionCacheInterface, startPosi
 		return errors.Trace(cache.Flush())
 	}
 
-	positionValues, err := Deserialize(position.Value)
-	if err != nil {
-		return errors.Trace(err)
+	positionValue, ok := position.Value.(*OplogPositionValue)
```

let it panic is enough I think